cmd/podman/pods/clone.go

```diff
@@ -0,0 +1,87 @@
+package pods
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/containers/common/pkg/completion"
+	"github.com/containers/podman/v4/cmd/podman/common"
+	"github.com/containers/podman/v4/cmd/podman/registry"
+	"github.com/containers/podman/v4/libpod/define"
+	"github.com/containers/podman/v4/pkg/domain/entities"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+var (
+	podCloneDescription = `Create an exact copy of a pod and the containers within it`
+
+	podCloneCommand = &cobra.Command{
+		Use:               "clone [options] POD NAME",
+		Short:             "Clone an existing pod",
+		Long:              podCloneDescription,
+		RunE:              clone,
+		Args:              cobra.RangeArgs(1, 2),
+		ValidArgsFunction: common.AutocompleteClone,
+		Example:           `podman pod clone pod_name new_name`,
+	}
+)
+
+var (
+	podClone entities.PodCloneOptions
+)
+
+func cloneFlags(cmd *cobra.Command) {
+	flags := cmd.Flags()
+
+	destroyFlagName := "destroy"
+	flags.BoolVar(&podClone.Destroy, destroyFlagName, false, "destroy the original pod")
+
+	startFlagName := "start"
+	flags.BoolVar(&podClone.Start, startFlagName, false, "start the new pod")
+
+	nameFlagName := "name"
+	flags.StringVarP(&podClone.CreateOpts.Name, nameFlagName, "n", "", "name the new pod")
+	_ = podCloneCommand.RegisterFlagCompletionFunc(nameFlagName, completion.AutocompleteNone)
+
+	common.DefineCreateFlags(cmd, &podClone.InfraOptions, true, false)
+	podClone.InfraOptions.MemorySwappiness = -1 // this is not implemented for pods yet, need to set -1 default manually
+
+	// need to fill an empty ctr create option for each container for sane defaults
+	// for now, these cannot be used. The flag names conflict too much
+	// this makes sense since this is a pod command not a container command
+	// TODO: add support for container specific arguments/flags
+	common.DefineCreateFlags(&cobra.Command{}, &podClone.PerContainerOptions, false, true)
```

Comment:
This looks very weird. Why do we need both PerContainerOptions and InfraOptions?

Reply:
Calling this just to init defaults adds unnecessary compute time and memory usage because it creates a complete new flag set on a stub command that is never used. If the defaults do not work we should add a function which properly initializes the correct defaults then.